xla/pjrt/c/pjrt_c_api_gpu_internal.cc

```diff
@@ -167,34 +167,101 @@ PJRT_Error* PJRT_ExecuteContext_Create(PJRT_ExecuteContext_Create_Args* args) {
   return nullptr;
 }
 
+namespace {
+
+struct TargetConfigAndDevices {
+  tensorflow::se::GpuTargetConfigProto target_config_proto;
+  std::vector<int> device_ids;
+};
+
+absl::StatusOr<TargetConfigAndDevices> GetTargetConfigFromOptions(
+    absl::flat_hash_map<std::string, xla::PjRtValueType>& options) {
+  tensorflow::se::GpuTargetConfigProto target_config_proto;
+  std::vector<int> device_ids;
+  std::string target_config_proto_string;
+  const auto& debug_options = xla::GetDebugOptionsFromFlags();
+
+  auto target_config_it = options.find("target_config");
+  if (target_config_it != options.end()) {
+    target_config_proto_string =
+        std::get<std::string>(target_config_it->second);
+    if (!tsl::protobuf::TextFormat::ParseFromString(target_config_proto_string,
+                                                    &target_config_proto)) {
+      return absl::FailedPreconditionError(
+          "Failed to parse GpuTargetConfigProto "
+          "from the 'target_config' parameter.");
+    }
+  } else if (!debug_options.xla_gpu_target_config_filename().empty()) {
```

Comment:
TBH reading XLA_FLAGS here feels like layering violation, should we be using PjRT options instead? IMO we should use one or another, not both